### PR TITLE
Update Metadata.java

### PR DESCRIPTION
**Pull request by davidgraeff**
Potential fix for #699 

Signed-off-by: David Graeff <david.graeff@web.de>

### CODE REVIEW

File: bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/items/Metadata.java [hunk 1]
```diff
@@ -25,15 +25,17 @@
...
 public final class Metadata implements Identifiable<MetadataKey> {
 
-    private final MetadataKey key;
-    private final String value;
-    private final Map<String, Object> configuration;
+    private MetadataKey key;
+    private String value;
+    private Map<String, Object> configuration;
+
+    Metadata() {
```
Comment (maggu2810) [inline bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/items/Metadata.java]: Don't remove `final`.
Add the comment that is (should be) used on all the other such constructors, too.

Comment (davidgraeff) [reply bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/items/Metadata.java]: I get you. Immutable classes are preferable. Is gson able to modify final fields?

Comment (maggu2810) [reply bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/items/Metadata.java]: Gson uses reflection to set final fields (via `.setAccessible(true)`),

**Comment by maggu2810** [comment]
Has this been closed by intention?
A PR tracks a branch, so no need to close and create a new PR as long as the branch keeps the same.

**Pull request closed without merging**